File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    let result = [0, 0];
    const max_x = (board[0] - 1) / 2;
    const max_y = (board[1] - 1) / 2;
    
    for (key of keyinput) {
        if (key === "left" && result[0] > -max_x) {
            result[0] -= 1;
        } else if (key === "right" && result[0] < max_x) {
            result[0] += 1;
        } else if (key === "down" && result[1] > -max_y) {
            result[1] -= 1;
        } else if (key === "up" && result[1] < max_y) {
            result[1] += 1;
        }
    }
    
    return result;
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    result = [0, 0]
    max_x = board[0] // 2
    max_y = board[1] // 2
    
    for key in keyinput:
        if key == "left" and result[0] > -max_x:
            result[0] -= 1
        elif key == "right" and result[0] < max_x:
            result[0] += 1
        elif key == "down" and result[1] > -max_y:
            result[1] -= 1
        elif key == "up" and result[1] < max_y:
            result[1] += 1
    
    return result